business/proxy_status.go

```diff
@@ -74,3 +75,50 @@ func xdsStatus(sent, acked string) string {
 	// Since the Nonce changes to uuid, so there is no more any time diff info
 	return "Stale"
 }
+
+func (in *ProxyStatus) GetConfigDump(namespace, pod string) (interface{}, error) {
+	var err error
+	promtimer := internalmetrics.GetGoFunctionMetric("business", "ProxyStatus", "GetConfigDump")
+	defer promtimer.ObserveNow(&err)
+
+	dump, err := in.k8s.GetConfigDump(namespace, pod)
+	if err != nil {
+		return nil, err
+	}
+
+	return dump, nil
+}
+
+func (in *ProxyStatus) GetConfigDumpResourceEntries(namespace, pod, resource string) (interface{}, error) {
```

Comment:
In this one, I'd split the methods with their own returning type.

Note, it's perfectly ok to have inside of the model.ConfigDump interfaces to not add all nested values in complex structures.

Reply:
I disagree in the fact to split methods just for returning specific types. I guess that just returning a common interface would be alright. I created an small interface for dealing with the summaries in the `buildDump` method right below. It makes it easier for removing the complexity of managing different type structures to achieve the very same result.

Using that, the method signature is:
`func (in *ProxyStatus) GetConfigDumpResourceEntries(namespace, pod, resource string) (models.ResourceDump, error)`